src/sentry/migrations/0180_copy_useroptions_to_notificationsettings.py

```diff
@@ -0,0 +1,154 @@
+                value = NotificationSettingOptionValues.ALWAYS
+            if user_option.value == UserOptionValue.participating_only:
+                value = NotificationSettingOptionValues.SUBSCRIBE_ONLY
+            if user_option.value == UserOptionValue.no_conversations:
+                value = NotificationSettingOptionValues.NEVER
+        elif user_option.key == "mail:alert":  # fine tuned project alerts
+            if user_option.project_id:
+                scope_identifier = user_option.project_id
+                scope_type = NotificationScopeType.PROJECT
+            else:
+                scope_identifier = user_option.user.id
+                scope_type = NotificationScopeType.USER
+            type = NotificationSettingTypes.ISSUE_ALERTS
+            if user_option.value == 0:
+                value = NotificationSettingOptionValues.NEVER
+            if user_option.value == 1:
+                value = NotificationSettingOptionValues.ALWAYS
+        elif user_option.key == "subscribe_by_default":  # top level project alerts on/off
+            scope_identifier = user_option.user.id
+            scope_type = NotificationScopeType.USER
+            type = NotificationSettingTypes.ISSUE_ALERTS
+            if user_option.value == 1:
+                value = NotificationSettingOptionValues.ALWAYS
+            if user_option.value == 0:
+                value = NotificationSettingOptionValues.NEVER
+        elif user_option.key == "deploy-emails":
+            # if you have fine tuning for an org, organization_id will have a value (rather than None)
+            if user_option.organization_id:
+                scope_identifier = user_option.organization_id
+            else:
+                scope_identifier = user_option.user.id
+            scope_type = NotificationScopeType.ORGANIZATION
+            type = NotificationSettingTypes.DEPLOY
+            # if you've not explicitly set anything OR set it to default, there is no db row
+            # by default deploy notifications are set to committed_deploys_only,
+            # but there will be an entry for the top level alert option
+            # if you change the value to something else
+            if user_option.value == UserOptionValue.all_deploys:
+                value = NotificationSettingOptionValues.ALWAYS
+            if user_option.value == UserOptionValue.no_deploys:
+                value = NotificationSettingOptionValues.NEVER
+            if user_option.value == UserOptionValue.committed_deploys_only:
+                value = NotificationSettingOptionValues.COMMITTED_ONLY
+        else:
+            continue
+        user = User.objects.select_related("actor").get(id=user_option.user_id)
+        NotificationSetting.objects.update_or_create(
+            scope_type=scope_type,  # user, org, or project
+            scope_identifier=scope_identifier,  # user_id, organization_id, or project_id
+            target=user,
```

Comment:
I think this still need to be `user.actor`

Reply:
oop you're right, re-read the docs